src/lib/api/entity-api.ts

```diff
@@ -419,7 +431,7 @@ export namespace EntityApi {
        * @returns Promise resolving to the created entity
        */
 
-      create: (entity: T) => executeWithState(() => api.create(entity)),
+      create: (entity: T) => executeWithState(() => api.create!(entity)),
```

Comment:
So `!` is necessary to suppress Typescript errors since you changed `create()` to be optionally defined with the use of `?` - do I have this right?

Reply:
yes that's my understanding as well, although it seems like a bit of a hack as well since we're essentially bypassing the type checks. Similar to my comment above, I think it may make sense to come back around and take improve how we're handling the typing on this part of the application. Probably not super important right this second.